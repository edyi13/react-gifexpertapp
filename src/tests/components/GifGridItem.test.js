import {shallow} from 'enzyme';
import { GifGridItem } from '../../Components/GifGridItem';


describe('Pruebas en el <GifGridItem/>', () => {
    //Global
    const title = 'Dragon Ball Z GIF';
    const url = 'http://gph.is/2cLjMwm';
    let wrapper = shallow(<GifGridItem title={title} url={url} />);
    beforeEach( () => {
        wrapper = shallow(<GifGridItem title={title} url={url} />);
    });
  
    test('Deberia de mostrar <GifGridItem/> correctamente', () => {
        //Arrange
        //Act
        //Assert
        expect(wrapper).toMatchSnapshot();
    })

    test('Deber de tener un parrafo con el title', () => {
        //Arrange
        
        //Act
        const p = wrapper.find('p');
        //Expect
        expect(p.text().trim()).toBe(title);
    })

    test('Debe tener la imagen igual url y alt de props', () => {
        //Arrange
        //Act
        const img = wrapper.find('img');
        //Assert
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    })
    
    test('Deberia de tener la clase animate_fadeIn', () => {
        //Arrange
        //Act
        const div =wrapper.find('div');
        //Assert
        expect(div.props().className).toContain('animate__fadeIn');

        //Forma de Fernando
        //Arrange
        //Act
        const className = div.prop('className');
        //Assert
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    
    
})
