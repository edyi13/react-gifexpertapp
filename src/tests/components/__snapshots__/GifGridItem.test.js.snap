// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGridItem/> Deberia de mostrar <GifGridItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__fadeIn"
>
  <img
    alt="Dragon Ball Z GIF"
    src="http://gph.is/2cLjMwm"
  />
  <p>
    Dragon Ball Z GIF
  </p>
</div>
`;
